commands/system/help.js

```diff
@@ -9,74 +11,170 @@ class Help extends Command {
       description: "Get help on a command, command category, or a setting",
       extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
       category: "System",
-      usage: "help <category/command/setting> [page-num]",
-      aliases: ["h", "halp", "commands"]
+      usage: "help <category/command/setting>",
+      aliases: ["h", "halp", "commands"],
     });
+    this.pages = async (message, msg2, page, sorted, type, level, r, this2, hm) => {
+      let n = 0;
+      sorted.forEach((c) => {
+        if (c.help.category.toLowerCase() === type.toLowerCase()) {
+          n++;
+        }
+      });
+
+      let num = 0;
+      let output = "";
+      const pg = Number(page) && Number(page) <= Math.ceil(n / perpage) ? Number(page) : 1;
+      for (const c of sorted.values()) {
+        if (c.help.category.toLowerCase() === type.toLowerCase()) {
+          if (c.help.cat === "NSFW" && !message.channel.nsfw) return;
```

Comment:
If I recall correctly, that was the point of that one there. It's one I put in a while ago to prevent NSFW categories to be posted in non-nsfw channels.

Reply:
Still, should use `continue;` so it shows the non-nsfw commands. Also, it's already filtered before:

https://github.com/YoussefElshemi/Misaki/blob/f3cb93b72f33315e3ad803f724b4277ba077039f/commands/system/help.js#L70

Reply:
If the category is NSFW, all the commands in that category **will** be NSFW...

Reply:
Then check `if (type.toLowerCase() === "nsfw")` is true, beforehand, instead of doing a lot of heavy processing and finally, check it.

Reply:
✔️